### PR TITLE
Add default extensions

### PR DESCRIPTION
**Pull request by schoersch**
Adds the option to specify the extensions domain independent or omit the specification fully load all default image types.

Note that I added a dataset with different extensions in a single class (mixed-codecs) for testing.

This should also resolve the coverage issue #63.

### CODE REVIEW

**Review by smilesun — CHANGES_REQUESTED** [review]
Could you revert the changes on the images (jpg files)?  Those files should be permanent there and they are binary files (instead of text files) so we do not want to make any modifications on them.

**Comment by schoersch** [comment]
> Could you revert the changes on the images (jpg files)? Those files should be permanent there and they are binary files (instead of text files) so we do not want to make any modifications on them.

I did not changed any of the existing files. However, in order to test if the filtering by extensions works, I added the mixed_codecs test dataset. Which adds a few additional permanent binary files.

But if you prefer, I can of course remove this dataset and the corresponding tests.

**Comment by smilesun** [comment]
@schoersch , ah sorry, I misread the path of the images you uploaded, if they are new files, fine for me!

**Review by smilesun — CHANGES_REQUESTED** [review]
Could you add 1-line docstring """text""" to the functions listed here so we know what each function is doing? 

https://app.codacy.com/gh/marrlab/DomainLab/pullRequest?prid=10989386

File: domainlab/dsets/dset_subfolder.py [hunk 1]
```diff
@@ -51,18 +62,24 @@ class DsetSubFolder(DatasetFolder):
...
-                             target_transform=target_transform,
-                             is_valid_file=is_valid_file)
+
+        if is_valid_file is not None and extensions is not None:
+            raise ValueError(
+                "Both extensions and is_valid_file cannot be not None at the same time")
+
+        if is_valid_file is None and extensions is None:
+            # setting default extensions
+            extensions = ('jpg', 'jpeg', 'png')
```
[the remark this answers is not in the document]
Comment (schoersch) [reply domainlab/dsets/dset_subfolder.py:72]: The question is if in this case we want to:
1. Read any image file, then we have to make an extensions tuple of all image extensions supported by PIL
2. Or we read any file in the directory. Then we can go for the all_true solution, but the code is crashing when we're stumbling across a non-image file. This is currently the case and reason while the test fails in #69 . I.e. deleting the file *data/mixed_codec/caltech/auto/text.txt* let the tests pass.

[the remark this answers is not in the document]
Comment (smilesun) [reply domainlab/dsets/dset_subfolder.py:72]: After some long time thinking, i  think it make sense to only raise a warning to the user that we have only 3 default file suffix. The user should be responsible to not to forget to include image files

**Comment by smilesun** [comment]
@aliechoes , @peterlienemann  I am done with my review from @schoersch , do you two wanna take a look as well?

**Review by smilesun — APPROVED** [review]

**Pull request merged**